openedx/features/content_type_gating/tests/test_partitions.py

```diff
@@ -93,7 +94,7 @@ def test_access_denied_fragment_for_masquerading(self):
             'crum.get_current_request',
             return_value=mock_request
         ):
-            fragment = partition.access_denied_fragment(mock_block, global_staff, 'test_group', 'test_allowed_group')
+            fragment = partition.access_denied_fragment(mock_block, global_staff, GroupFactory(), 'test_allowed_group')
```

Comment:
Do we need to do anything to make sure this case is tested? Or did our existing tests somehow already test it?

Reply:
this was just a failing test, yeah i can add a new test specifically for the limited access group content case, i first wanted to make sure the high level approach was correct

Reply:
i've added a new test_access_denied_fragment_for_full_access_users test